File: RobotBehavior/Arbiter/Component/Led/LedBrightness.py

from RobotBehavior.Arbiter.Action.ArbiterAction import ArbiterAction


class LedBrightness(ArbiterAction):
	def __init__(self, arbiter_config):
		self._brightness = 0
		arbiter_config['name'] = "change LED brightness"
		super()._init__(arbiter_config)


	def get_brightness(self):
		return self._brightness

	def set_brightness(self, brightness):
		self._brightness = brightness
	

File: RobotBehavior/Arbiter/Component/Led/LedBrightness.java
package de.hska.lat.behavior.arbiter.component.led;


import de.hska.lat.behavior.arbiter.action.ArbiterAction;

public class LedBrightness extends ArbiterAction
{

	
	protected float brightness;


	private static final String name = "change led brightness";
	
	
	public LedBrightness(int priority)
	{
		super(priority);
	}

	@Override
	public String getName()
	{
		return(LedBrightness.name);
	}


	@Override
	public String toString ()
	{
		String returnString;
		
		returnString = this.getName();
		
		return(returnString);
	}
	
	public float getBrightness()
	{
		return brightness;
	}

	public void setBrightness(float brightness)
	{
		this.brightness = brightness;
	}

}
